server/src/main/java/org/elasticsearch/rest/action/cat/RestSnapshotAction.java

```diff
@@ -102,29 +107,41 @@ protected Table getTableWithHeader(RestRequest request) {
 
     private Table buildTable(RestRequest req, GetSnapshotsResponse getSnapshotsResponse) {
         Table table = getTableWithHeader(req);
-        for (SnapshotInfo snapshotStatus : getSnapshotsResponse.getSnapshots()) {
-            table.startRow();
-
-            table.addCell(snapshotStatus.snapshotId().getName());
-            table.addCell(snapshotStatus.state());
-            table.addCell(TimeUnit.SECONDS.convert(snapshotStatus.startTime(), TimeUnit.MILLISECONDS));
-            table.addCell(FORMATTER.format(Instant.ofEpochMilli(snapshotStatus.startTime())));
-            table.addCell(TimeUnit.SECONDS.convert(snapshotStatus.endTime(), TimeUnit.MILLISECONDS));
-            table.addCell(FORMATTER.format(Instant.ofEpochMilli(snapshotStatus.endTime())));
-            final long durationMillis;
-            if (snapshotStatus.state() == SnapshotState.IN_PROGRESS) {
-                durationMillis = System.currentTimeMillis() - snapshotStatus.startTime();
-            } else {
-                durationMillis = snapshotStatus.endTime() - snapshotStatus.startTime();
+
+        if (getSnapshotsResponse.isFailed()) {
+            throw new ElasticsearchException(
```

Comment:
this hides the original exception?

Reply:
Indeed, but cat API is not well-suited for reporting multiple exceptions. I think in this case, /_snapshot/... API could be used to get exact reasons why repositories are failing.

Reply:
why not add the original exceptions as cause?

Reply:
@ywelsch do you mean something like this https://github.com/elastic/elasticsearch/pull/42090/commits/f19ff8f3c6372928d71c50cf0f4f9858dc3c93c4 ?

Reply:
yes